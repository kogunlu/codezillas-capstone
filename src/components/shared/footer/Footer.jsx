import { Link } from 'react-router-dom';
import { FaFacebook, FaTwitter } from 'react-icons/fa';
import { GrGoogle } from 'react-icons/gr';
import SubsInput from '../subsInput/SubsInput';

function Footer() {
  return (
    <div className="w-full h-60 md:h-44 bg-amber-200 flex justify-center ">
      <div className="w-full md:w-10/12 h-full flex flex-col md:flex-row justify-between items-center">
        <div className="h-5/6 w-5/6 md:w-4/12 flex flex-col items-center md:items-start justify-around">
          <div className="flex flex-col items-center md:items-start gap-2 w-full ">
            <h2 className="text-2xl font-semibold">Subscribe</h2>
            <p className="text-gray-500 ">
              We`ll never to spam you or share your email
            </p>
          </div>

          <SubsInput />
        </div>

        <div className="h-5/6 w-full md:w-4/12 flex flex-col justify-evenly items-center">
          <div className="w-full flex justify-between items-center gap-2">
            <Link
              to="/"
              className="text-gray-500 hover:scale-110 hover:underline hover:underline-offset-4 text-lg"
            >
              Home
            </Link>
            <Link
              to="blog"
              className="text-gray-500 hover:scale-110 hover:underline hover:underline-offset-4 text-lg"
            >
              Blogs
            </Link>
            <Link
              to="about"
              className="text-gray-500 hover:scale-110 hover:underline hover:underline-offset-4 text-lg"
            >
              About
            </Link>
            <Link
              to="contact"
              className="text-gray-500 hover:scale-110 hover:underline hover:underline-offset-4 text-lg"
            >
              Contact
            </Link>
          </div>

          <div className="w-6/12 flex justify-around items-center">
            <button
              type="button"
              alt="Twitter"
              className="text-3xl hover:text-cyan-400"
            >
              <FaTwitter />
            </button>
            <button
              type="button"
              alt="Facebook"
              className="text-3xl hover:text-blue-800"
            >
              <FaFacebook />
            </button>
            <button
              type="button"
              alt="Google"
              className="text-3xl hover:text-white"
            >
              <GrGoogle />
            </button>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Footer;
