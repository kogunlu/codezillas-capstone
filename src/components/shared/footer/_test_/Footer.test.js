import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom"
import Footer from "../Footer";

it('Renders correctly for Footer Component', () => {
    const tree = renderer
    .create(
    <BrowserRouter>
    <Footer />
    </BrowserRouter> 
    )
    .toJSON();
    expect(tree).toMatchSnapshot();
  }); 