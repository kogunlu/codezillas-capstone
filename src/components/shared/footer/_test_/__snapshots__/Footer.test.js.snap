// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly for Footer Component 1`] = `
<div
  className="w-full h-60 md:h-44 bg-amber-200 flex justify-center "
>
  <div
    className="w-full md:w-10/12 h-full flex flex-col md:flex-row justify-between items-center"
  >
    <div
      className="h-5/6 w-4/12 flex flex-col items-center md:items-start justify-around"
    >
      <div
        className="flex flex-col gap-2 "
      >
        <h2
          className="text-2xl font-semibold"
        >
          Subscribe
        </h2>
        <p
          className="text-gray-500 "
        >
          We\`ll never to spam you or share your email
        </p>
      </div>
      <div
        className="flex justify-start items-center"
      >
        <input
          className="border border-gray-400 rounded-l-md h-10 pl-2 focus:outline-none "
          onChange={[Function]}
          placeholder="Enter your e-mail"
          type="email"
          value=""
        />
        <button
          className="h-10  bg-cyan-400 pl-1 pr-2 rounded-r-md border-y border-r border-gray-400 hover:bg-cyan-300 text-xl"
          onClick={[Function]}
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M446.7 98.6l-67.6 318.8c-5.1 22.5-18.4 28.1-37.3 17.5l-103-75.9-49.7 47.8c-5.5 5.5-10.1 10.1-20.7 10.1l7.4-104.9 190.9-172.5c8.3-7.4-1.8-11.5-12.9-4.1L117.8 284 16.2 252.2c-22.1-6.9-22.5-22.1 4.6-32.7L418.2 66.4c18.4-6.9 34.5 4.1 28.5 32.2z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="h-5/6 w-full md:w-4/12 flex flex-col justify-evenly items-center"
    >
      <div
        className="w-full flex justify-between items-center gap-2"
      >
        <a
          className="text-gray-500 hover:scale-110 hover:underline hover:underline-offset-4 text-lg"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="text-gray-500 hover:scale-110 hover:underline hover:underline-offset-4 text-lg"
          href="/blogs"
          onClick={[Function]}
        >
          Blogs
        </a>
        <a
          className="text-gray-500 hover:scale-110 hover:underline hover:underline-offset-4 text-lg"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          className="text-gray-500 hover:scale-110 hover:underline hover:underline-offset-4 text-lg"
          href="/contact"
          onClick={[Function]}
        >
          Contact
        </a>
      </div>
      <div
        className="w-6/12 flex justify-around items-center"
      >
        <button
          alt="Twitter"
          className="text-3xl hover:text-cyan-400"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
            />
          </svg>
        </button>
        <button
          alt="Facebook"
          className="text-3xl hover:text-blue-800"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
            />
          </svg>
        </button>
        <button
          alt="Google"
          className="text-3xl hover:text-white"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.26620003,9.76452941 C6.19878754,6.93863203 8.85444915,4.90909091 12,4.90909091 C13.6909091,4.90909091 15.2181818,5.50909091 16.4181818,6.49090909 L19.9090909,3 C17.7818182,1.14545455 15.0545455,0 12,0 C7.27006974,0 3.1977497,2.69829785 1.23999023,6.65002441 L5.26620003,9.76452941 Z"
            />
            <path
              d="M16.0407269,18.0125889 C14.9509167,18.7163016 13.5660892,19.0909091 12,19.0909091 C8.86648613,19.0909091 6.21911939,17.076871 5.27698177,14.2678769 L1.23746264,17.3349879 C3.19279051,21.2936293 7.26500293,24 12,24 C14.9328362,24 17.7353462,22.9573905 19.834192,20.9995801 L16.0407269,18.0125889 Z"
            />
            <path
              d="M19.834192,20.9995801 C22.0291676,18.9520994 23.4545455,15.903663 23.4545455,12 C23.4545455,11.2909091 23.3454545,10.5272727 23.1818182,9.81818182 L12,9.81818182 L12,14.4545455 L18.4363636,14.4545455 C18.1187732,16.013626 17.2662994,17.2212117 16.0407269,18.0125889 L19.834192,20.9995801 Z"
            />
            <path
              d="M5.27698177,14.2678769 C5.03832634,13.556323 4.90909091,12.7937589 4.90909091,12 C4.90909091,11.2182781 5.03443647,10.4668121 5.26620003,9.76452941 L1.23999023,6.65002441 C0.43658717,8.26043162 0,10.0753848 0,12 C0,13.9195484 0.444780743,15.7301709 1.23746264,17.3349879 L5.27698177,14.2678769 Z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
